### ISSUE TITLE
Panic in UPnP search

### ISSUE THREAD

**Issue by deffrian** [sigp/lighthouse#4171]
## Description
Lighthouse crashes during startup

## Version
a53830fd60a119bf3f659b253360af8027128e83 (current master)

## Present Behaviour
`lighthouse bn --network sepolia --execution-endpoint http://localhost:8551 --execution-jwt <path> --checkpoint-sync-url https://sepolia.checkpoint-sync.ethpandaops.io --disable-deposit-contract-sync`

```
Apr 06 13:19:14.045 INFO Execution payloads are pruned           service: freezer_db
thread 'tokio-runtime-worker' panicked at 'invalid url or method: Error(InvalidBaseUrl)', /home/deffrian/.cargo/registry/src/github.com-1ecc6299db9ec823/attohttpc-0.10.1/src/request.rs:94:41
stack backtrace:
   0: rust_begin_unwind
             at /rustc/2c8cc343237b8f7d5a3c3703e3a87f2eb2c54a74/library/std/src/panicking.rs:575:5
   1: core::panicking::panic_fmt
             at /rustc/2c8cc343237b8f7d5a3c3703e3a87f2eb2c54a74/library/core/src/panicking.rs:64:14
   2: core::result::unwrap_failed
             at /rustc/2c8cc343237b8f7d5a3c3703e3a87f2eb2c54a74/library/core/src/result.rs:1790:5
   3: attohttpc::request::RequestBuilder::new
   4: igd::search::search_gateway
   5: network::nat::construct_upnp_mappings
   6: tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
   7: tokio::runtime::task::core::Core<T,S>::poll
   8: tokio::runtime::task::harness::Harness<T,S>::poll
   9: tokio::runtime::blocking::pool::Inner::run
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
Apr 06 13:19:14.066 CRIT Task panic. This is a bug!              advice: Please check above for a backtrace and notify the developers, message: <none>, task_name: UPnP
Apr 06 13:19:14.066 INFO Internal shutdown received              reason: Panic (fatal error)
Apr 06 13:19:14.066 INFO Shutting down..                         reason: Failure("Panic (fatal error)")
Apr 06 13:19:14.067 INFO Saved DHT state                         service: network
Apr 06 13:19:14.067 INFO Network service shutdown                service: network
Apr 06 13:19:14.067 INFO HTTP server is disabled
Apr 06 13:19:14.068 INFO Saved beacon chain to disk              service: beacon
Panic (fatal error)
```

## Expected Behaviour
No exception

**Comment by deffrian** [comment]
also on 4d17fb3

**Comment by michaelsproul** [comment]
Which OS? Any virtualisation or containerisation in play? Looks like a weird error from the OS being unable to do UPnP.

You can probably work around it with `--disable-upnp`

**Comment by deffrian** [comment]
Thanks, works with `--disable-upnp`. Os: Ubuntu 22.04.2

**Comment by deffrian** [comment]
No virtualization or containers

**Comment by michaelsproul** [comment]
Our UPnP support is best-effort due to the vast number of routers and networking setups that are possible, but I think we could avoid panicking in this case (a clean failure would be better).

Which router do you have, if you don't mind me asking? You can also DM the answer on Discord, I'm `@sproul#3907`, you can cross-check my profile on the Lighthouse Discord.

**Comment by michaelsproul** [comment]
In terms of a fix, looks like maybe we'd have to go to the `igd` dep.

**Comment by deffrian** [comment]
I don't know it was on public wi-fi. I will try to ask owner of router in few days

**Comment by ackintosh** [comment]
I had a look at the `igd` a bit, perhaps the PR below might improve this issue:

https://github.com/sbstp/rust-igd/pull/55

**Comment by ackintosh** [comment]
I have filed a PR to rust-igd. https://github.com/sbstp/rust-igd/pull/56

**Comment by deffrian** [comment]
Sorry, I can't reproduce it anymore. Just prints
 `INFO UPnP not available    error: IO error: Resource temporarily unavailable (os error 11), service: UPnP`
 On commit 4d17fb3af6eeb37f4cd2f8c79873700e7d0d6587

**Comment by ackintosh** [comment]
In current `unstable` branch, we have updated the dependency causing the panic. Now, if there's any issue while searching for a gateway, lighthouse will emit logs indicating what is wrong.

**Comment by divagant-martian** [comment]
Closed by #4257
